services/src/main/java/org/fao/geonet/api/records/MetadataInsertDeleteApi.java

```diff
@@ -265,6 +269,9 @@ public SimpleMetadataProcessingReport deleteRecords(
                     MetadataUtils.backupRecord(metadata, context);
                 }
 
+                MetadataPreRemove preRemoveEvent = new MetadataPreRemove(metadata);
```

Comment:
If this fails then there will still be a backup record created.
I think this needs to be moved up before the backup.

Reply:
Good point. I will move it before the backup operation.